scripts/merge-script/strategies/api-merge.ts

```diff
@@ -0,0 +1,191 @@
+    if (failure !== null) {
+      return failure;
+    }
+
+    const mergeOptions: PullsMergeParams = {
+      pull_number: prNumber,
+      merge_method: method,
+      ...this.git.repoParams,
+    };
+
+    // If a commit message fixup is needed and we are merging using squash, we prompt
+    // for commit message changes. Note that this does not work with other merge methods
+    // as the Github API only allows commit message modifications for squash merging.
+    if (needsCommitMessageFixup && method === 'squash') {
+      await this._promptCommitMessageEdit(pullRequest, mergeOptions);
+    }
+
+    // Merge the pull request using the Github API into the selected base branch.
+    const {status, data: {sha}} = await this.git.api.pulls.merge(mergeOptions);
+
+    // https://developer.github.com/v3/pulls/#response-if-merge-cannot-be-performed
+    // Pull request cannot be merged due to merge conflicts.
+    if (status === 405) {
+      return {status: PullRequestStatus.MERGE_CONFLICTS, context: [githubTargetBranch]};
+    }
+    if (status !== 200) {
+      return {status: PullRequestStatus.UNKNOWN_MERGE_ERROR};
+    }
+
+    // If the PR does  not need to be merged into any other target branches,
+    // we exit here as we already completed the merge.
+    if (!cherryPickTargetBranches.length) {
+      return null;
+    }
+
+    // Refresh the target branch the PR has been merged into through the API. We need
+    // to re-fetch as otherwise we cannot cherry-pick the new commits into the remaining
+    // target branches.
+    this.fetchTargetBranches([githubTargetBranch]);
+
+    // Amount of commits that have landed in the target branch. This could vary from
+    // the amount of commits in the PR due to squashing.
+    const targetCommitsAmount = method === 'squash' ? 1 : pullRequest.commitsAmount;
+
+    // Cherry pick the merged commits into the remaining target branches.
+    const failedBranches = await this.cherryPickIntoTargetBranches(
+        `${sha}~${targetCommitsAmount}..${sha}`, cherryPickTargetBranches);
+
+    if (failedBranches.length) {
+      return {status: PullRequestStatus.MERGE_CONFLICTS, context: failedBranches};
+    }
+
+    this.pushTargetBranchesUpstream(cherryPickTargetBranches);
+    return null;
+  }
+
+  /**
+   * Prompts the user for the commit message changes. Unlike as in the autosquash merge
+   * strategy, we cannot start an interactive rebase because we merge using the Github API.
+   * The Github API only allows modifications to PR title and body for squash merges.
+   */
+  async _promptCommitMessageEdit(pullRequest: PullRequest, mergeOptions: PullsMergeParams) {
+    const commitMessage = await this._getDefaultSquashCommitMessage(pullRequest);
+    const {result} = await prompt<{result: string}>({
+      type: 'editor',
+      name: 'result',
+      message: 'Please update the commit message',
+      default: commitMessage,
+    });
+
```

Comment:
Should we also be validating the commit message that is created here?

Reply:
I'd say yes, but that seems to be something we can tackle as a follow-up? Especially since we are not having commit message validation in the components repo yet. I'd like to set it up soon, but I think we need some upstream changes to account for a different `scope` that we support.

Reply:
SGTM